fluid/PaddleCV/image_classification/dist_train/dist_train.py

```diff
@@ -28,127 +30,115 @@
-    parser.add_argument(
-        '--data_dir',
-        type=str,
-        default="../data/ILSVRC2012",
-        help="The ImageNet dataset root dir."
-    )
+    parser = argparse.ArgumentParser(description=__doc__)
+    add_arg = functools.partial(add_arguments, argparser=parser)
+    # yapf: disable
+    add_arg('batch_size',       int,   256,                  "Minibatch size.")
+    add_arg('use_gpu',          bool,  True,                 "Whether to use GPU or not.")
+    add_arg('total_images',     int,   1281167,              "Training image number.")
+    add_arg('num_epochs',       int,   120,                  "number of epochs.")
+    add_arg('class_dim',        int,   1000,                 "Class number.")
+    add_arg('image_shape',      str,   "3,224,224",          "input image size")
+    add_arg('model_save_dir',   str,   "output",             "model save directory")
+    add_arg('with_mem_opt',     bool,  False,                 "Whether to use memory optimization or not.")
+    add_arg('pretrained_model', str,   None,                 "Whether to use pretrained model.")
+    add_arg('checkpoint',       str,   None,                 "Whether to resume checkpoint.")
+    add_arg('lr',               float, 0.1,                  "set learning rate.")
+    add_arg('lr_strategy',      str,   "piecewise_decay",    "Set the learning rate decay strategy.")
+    add_arg('model',            str,   "DistResNet",         "Set the network to use.")
+    add_arg('enable_ce',        bool,  False,                "If set True, enable continuous evaluation job.")
+    add_arg('data_dir',         str,   "./data/ILSVRC2012",  "The ImageNet dataset root dir.")
+    add_arg('model_category',   str,   "models",             "Whether to use models_name or not, valid value:'models','models_name'" )
+    add_arg('fp16',             bool,  False,                "Enable half precision training with fp16." )
+    add_arg('scale_loss',       float, 1.0,                  "Scale loss for fp16." )
+    # for distributed
+    add_arg('update_method',      str,  "local",            "Can be local, pserver, nccl2.")
+    add_arg('multi_batch_repeat', int,  1,                  "Batch merge repeats.")
+    add_arg('start_test_pass',    int,  0,                  "Start test after x passes.")
+    add_arg('num_threads',        int,  8,                  "Use num_threads to run the fluid program.")
+    add_arg('split_var',          bool, True,               "Split params on pserver.")
+    add_arg('async_mode',         bool, False,              "Async distributed training, only for pserver mode.")
+    add_arg('reduce_strategy',    str,  "allreduce",        "Choose from reduce or allreduce.")
+    # yapf: enable
     args = parser.parse_args()
     return args
 
-def get_model(args, is_train, main_prog, startup_prog):
-    pyreader = None
-    class_dim = 1000
-    if args.data_format == 'NCHW':
-        dshape = [3, 224, 224]
+def get_device_num():
```

Comment:
add `os.getenv('FLAGS_selected_gpus')` to run with multi-processes mode.

Reply:
Should not change this here, it should always return actual num of devices on current node.